scripts/mkcloud

```diff
@@ -1304,7 +1304,7 @@ function sanity_checks
             Make sure to call mkcloud from your git clone."
     fi
 
-    if [ -z "$cloudsource" ] ; then
+    if [ -z "$cloudsource" ] && [[ ! $wantedcmds =~ "setuphost" ]] ; then
```

Comment:
So the sanity check should be skipped if the list of steps to be run contains "setuphost"?
Why? This does not make sense.
At least thats not what the PR summary says.

Reply:
You might mean:
```
... && [[ ! $wantedcmds = "setuphost" ]]
````

Reply:
I was following the documentation regarding [mkcloud setup](https://github.com/SUSE-Cloud/automation/blob/master/docs/mkcloud.md#mkcloud) and the `setuphost` step is called without any extra arguments when in fact `cloudsource` is required. I think it makes sense to remove that restriction to make it work according to documentation.

The check was imitating the ones for `setuplonelynodes` and `crowbar_register` ([link](https://github.com/SUSE-Cloud/automation/blob/master/scripts/mkcloud#L1339)), but I agree it makes more sense to have a strict checking.

Reply:
better edit line 1472 which already covers 3 others:
```bash
if [[ $steplist =~ ^(cleanup|help|steps)$ ]] ; then
```

Reply:
that's indeed a far better approach. Implementing that.

Reply:
@bmwiedemann 

But then you skip "all" sanity checks. Thats also not what is wanted here. Abel only wants to skip the check for the cloudsource variable in this specific case.
I think that line 1472 is not a good place for this change.